src/platform/ESP32/nimble/ChipDeviceScanner.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#if CONFIG_ENABLE_ESP32_BLE_CONTROLLER
+
+#include "blecent.h"
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/ESP32/ChipDeviceScanner.h>
+
+#define CHIPoBLE_SERVICE_UUID 0xFFF6
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+namespace {
+
+/// Retrieve CHIP device identification info from the device advertising data
+bool NimbleGetChipDeviceInfo(const ble_hs_adv_fields & fields, chip::Ble::ChipBLEDeviceIdentificationInfo & deviceInfo)
+{
+    // Check for CHIP Service UUID
+
+    if (fields.svc_data_uuid16 != NULL)
+    {
+        if (fields.svc_data_uuid16_len > 8 && fields.svc_data_uuid16[0] == 0xf6 && fields.svc_data_uuid16[1] == 0xff)
```

Comment:
This length check "> 8" is wrong (off by one) since it reads at position 9. Additionally, this will fail if there are multiple "Service Data - 16 bit UUID" fields in the advertisement data (e.g. another one for a different uuid), since the `svc_data_uuid16` will only contain a pointer to the last occurrence.